test/unit/playlist-test.js

```diff
@@ -58,4 +59,48 @@ describe('playlist', function() {
             }
         });
     });
+
+    describe('normalizePlaylistItem', function () {
+        let item;
+        let model;
+        beforeEach(function () {
+            model = new MockModel();
+            model.setup({});
+            item = new Item({
```

Comment:
This made me realize that we should be calling `Item` (from 'playlist/item') in `setPlaylistItem`. @DanFerrer can you add that to this PR?

In controller.js the item input could be a plain object and missing various properties. The playlist/item module exports a `Item` function that cleans up tracks and sources:

```js
        this.setPlaylistItem = function (index, item) {
            item = normalizePlaylistItem(_model, Item(item), item.feedData || {});
```
`Item` always returns a playlist object - even if it has no sources, `normalizePlaylistItem` will do additional source selection and return undefined if there are no sources present. It's split into these two methods to separate object creation of item filtering.

Reply:
Looks good now. Thanks!